### ISSUE TITLE
Multiple kinds of mv-if?

### ISSUE THREAD

**Issue by LeaVerou** [mavoweb/mavo#306]
While `mv-if` is useful, I have found that its defaults are not always desirable. There are multiple parameters on what kind of conditional is needed:

#### Parameter 1: Different behavior on read/edit mode

Often, you want the conditional to only apply on read mode, otherwise you cannot modify the data it depends on! Consider the following code: 

```html
<p class="comment" mv-if="feedback">
    Some text that should only show if there's feedback 
    <a property="feedback" target="_blank">Feedback</a>
</p>
```

As `mv-if` currently works, it’s impossible to enter a feedback link with the case above.

Also, often we want text that **only** appears in edit mode, to assist with editing.

#### Parameter 2: Does it affect the data? (yes/no/expressions only)

Currently, `mv-if` hides the data from expressions, but includes them in the stored data. 
This allows things like filtering, pagination etc without corrupting the data.
However, there are cases where you want entirely presentational hiding, as well as cases where you want the data to not be stored either (The [SVG Path builder](https://mavo.io/demos/svgpath/) is a good example of the latter).
The former can be done with CSS + expressions, but the latter is currently impossible.

#### Parameter 3: Does it affect the DOM? (yes/no)

Currently, `mv-if` removes the element from the DOM entirely. However, especially in cases where it's presentational, we often only want hiding via CSS or the `hidden` attribute. This is possible already via expressions in the `class`, `style`, or `hidden` attributes, but would be more elegant with an `mv-if`-esque attribute.

#### What to do?

So, basically we need 3 * 3 * 2 = 18 combinations of parameters. We can probably reduce that to 3 * 3 + 3 = 12 combinations, since it doesn't make sense to not remove from the DOM when the hiding is semantic.

Some options I see are:

- Introduce new attributes for each behavior, possibly in the form of `mv-if-*`. Kinda awkward and long IMO.
- Introduce a second attribute, `mv-if-options` with a space separated list of identifiers. E.g. `mv-if-options="read-only preserve-data presentational"` (names TBB)

Thoughts?

**Comment by LeaVerou** [comment]
@karger do see the edits before responding.

**Comment by LeaVerou** [comment]
We discussed this in our weekly F2F meeting today, and while we didn't reach a resolution, these are some notes from the discussion:

- Current mode (parameter 1) should probably be accessible to expressions instead of being an `mv-if` option.
- If the hiding is entirely presentational, it can be done via CSS already, so that means Parameter 3 is not that useful.
- It's unclear if there are any use cases where storing data hidden via `if` causes any problem other than bloating the data file.
- It's unclear if there are any use cases where wanting `mv-if` to **not** affect data where hiding via CSS is not an option. Perhaps to avoid hidden elements from messing with CSS selectors?
